### PR TITLE
ci: support `prefer-optional-chain`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-editor/components/toolbar/toolbar.component.ts [hunk 1]
```diff
@@ -186,9 +186,7 @@ export class TuiToolbarComponent {
             ({nativeFocusableElement}) => !!nativeFocusableElement,
         );
 
-        return first && first.nativeFocusableElement
-            ? first.nativeFocusableElement
-            : null;
+        return first?.nativeFocusableElement ? first.nativeFocusableElement : null;
```
Comment (waterplea) [inline projects/addon-editor/components/toolbar/toolbar.component.ts:189]: first?.nativeFocusableElement || null

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts [hunk 2]
```diff
@@ -33,19 +33,15 @@ export class TuiMobileCalendarDialogComponent {
     }
 
     get min(): TuiDay {
-        return this.context.data && this.context.data.min
-            ? this.context.data.min
-            : TUI_FIRST_DAY;
+        return this.context.data?.min ? this.context.data.min : TUI_FIRST_DAY;
```
Comment (waterplea) [inline projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts:36]: this.context.data?.min || TUI_FIRST_DAY

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts [hunk 3]
```diff
@@ -33,19 +33,15 @@ export class TuiMobileCalendarDialogComponent {
...
     }
 
     get max(): TuiDay {
-        return this.context.data && this.context.data.max
-            ? this.context.data.max
-            : TUI_LAST_DAY;
+        return this.context.data?.max ? this.context.data.max : TUI_LAST_DAY;
```
Comment (waterplea) [inline projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts:40]: Same

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts [hunk 4]
```diff
@@ -33,19 +33,15 @@ export class TuiMobileCalendarDialogComponent {
...
     }
 
     get disabledItemHandler(): TuiBooleanHandler<TuiDay> {
-        return this.context.data && this.context.data.disabledItemHandler
+        return this.context.data?.disabledItemHandler
```
Comment (waterplea) [inline projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts:44]: Same

File: projects/demo/src/modules/components/hosted-dropdown/examples/1/index.ts [hunk 5]
```diff
@@ -21,8 +21,6 @@ export class TuiHostedDropdownExample1 {
     onClick(): void {
         this.open = false;
```
Comment (waterplea) [inline projects/demo/src/modules/components/hosted-dropdown/examples/1/index.ts:23]: Unnecessary empty line now

File: projects/demo/src/modules/components/hosted-dropdown/examples/2/index.ts [hunk 6]
```diff
@@ -25,8 +25,6 @@ export class TuiHostedDropdownExample2 {
     onClick(): void {
         this.open = false;
```
Comment (waterplea) [inline projects/demo/src/modules/components/hosted-dropdown/examples/2/index.ts:27]: Here as well

File: projects/testing/utils/active-element.ts [hunk 7]
```diff
@@ -13,7 +13,7 @@ export function dispatchOnActive<T>(key: string, fixture?: ComponentFixture<T>):
 }
 
 export function activeText(): string {
-    return document.activeElement && document.activeElement.textContent
+    return document.activeElement?.textContent
```
Comment (waterplea) [inline projects/testing/utils/active-element.ts:16]: document.activeElement?.textContent?.trim() || ''

**Review by waterplea — CHANGES_REQUESTED** [review]
A few small comments

File: projects/addon-editor/components/toolbar/toolbar.component.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/components/toolbar/toolbar.component.ts]: fixed

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts]: fixed

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts]: fixed

File: projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-mobile/components/mobile-calendar-dialog/mobile-calendar-dialog.component.ts]: fixed

File: projects/demo/src/modules/components/hosted-dropdown/examples/1/index.ts (hunk 5 above)
Comment (splincode) [reply projects/demo/src/modules/components/hosted-dropdown/examples/1/index.ts]: fixed

File: projects/demo/src/modules/components/hosted-dropdown/examples/2/index.ts (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/components/hosted-dropdown/examples/2/index.ts]: fixed

File: projects/testing/utils/active-element.ts (hunk 7 above)
Comment (splincode) [reply projects/testing/utils/active-element.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**